extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -183,7 +199,93 @@ export class ResourceTypePickerDialog extends DialogBase {
 		this._dialogObject.content = [tab];
 	}
 
+	private createTagsListView(): azdata.ListViewComponent {
+
+		const tags = this.getAllResourceTags();
+		if (!tags.includes('All')) {
+			tags.splice(0, 0, 'All');
+		}
+		const items: azdata.ListViewOption[] = [];
+		tags.forEach((t: string, idx: number) => {
+			items.push({
+				label: localize('deploymentDialog.tag' + idx, '{0}', t),
+				id: t
+			});
+		});
+		const listView = this._view.modelBuilder.listView().withProps({
+			title: {
+				text: localize('deploymentDialog.tagsListViewTitle', 'Categories')
+			},
+			CSSStyles: {
+				'width': '140px',
+				'margin-top': '35px'
+			},
+			options: items,
+			selectedOptionId: items[0].id
+		}).component();
+		this._toDispose.push(listView.onDidClick((e) => {
+			this.filterResources();
+		}));
+
+		return listView;
+	}
+
+	private filterResources(): void {
+
+		const search = this._resourceSearchBox.value?.toLowerCase() ?? '';
+
+		const tag = this._resourceTagsListView.selectedOptionId!;
+
+		let filteredResourceTypes: ResourceType[] = [];
+
+		if (tag !== 'All') {
+			this._resourceTypes.forEach(element => {
+				if (element.tags?.includes(tag)) {
+					filteredResourceTypes.push(element);
+				}
+			});
+		} else {
+			filteredResourceTypes = this._resourceTypes;
+		}
+
+		let filteredResourceTypesonSearch: ResourceType[] = [];
+		filteredResourceTypes.forEach(element => {
```

Comment:
nit: conver to a 'map' call on filteredResourceTypes instead of forEach, that way filteredResourceTypesOnSearch not needed to be initialized with empty array.